src/SamplesApp/UITests.Shared/Windows_Devices/Midi/MidiDeviceInput.xaml.cs

```diff
@@ -0,0 +1,250 @@
+			// Build the received MIDI message into a readable format
+			StringBuilder outputMessage = new StringBuilder();
+			outputMessage.Append(receivedMidiMessage.Timestamp.ToString()).Append(", Type: ").Append(receivedMidiMessage.Type);
+
+			// Add MIDI message parameters to the output, depending on the type of message
+			switch (receivedMidiMessage.Type)
+			{
+				case MidiMessageType.NoteOff:
+					var noteOffMessage = (MidiNoteOffMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(noteOffMessage.Channel).Append(", Note: ").Append(noteOffMessage.Note).Append(", Velocity: ").Append(noteOffMessage.Velocity);
+					break;
+				case MidiMessageType.NoteOn:
+					var noteOnMessage = (MidiNoteOnMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(noteOnMessage.Channel).Append(", Note: ").Append(noteOnMessage.Note).Append(", Velocity: ").Append(noteOnMessage.Velocity);
+					break;
+				case MidiMessageType.PolyphonicKeyPressure:
+					var polyphonicKeyPressureMessage = (MidiPolyphonicKeyPressureMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(polyphonicKeyPressureMessage.Channel).Append(", Note: ").Append(polyphonicKeyPressureMessage.Note).Append(", Pressure: ").Append(polyphonicKeyPressureMessage.Pressure);
+					break;
+				case MidiMessageType.ControlChange:
+					var controlChangeMessage = (MidiControlChangeMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(controlChangeMessage.Channel).Append(", Controller: ").Append(controlChangeMessage.Controller).Append(", Value: ").Append(controlChangeMessage.ControlValue);
+					break;
+				case MidiMessageType.ProgramChange:
+					var programChangeMessage = (MidiProgramChangeMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(programChangeMessage.Channel).Append(", Program: ").Append(programChangeMessage.Program);
+					break;
+				case MidiMessageType.ChannelPressure:
+					var channelPressureMessage = (MidiChannelPressureMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(channelPressureMessage.Channel).Append(", Pressure: ").Append(channelPressureMessage.Pressure);
+					break;
+				case MidiMessageType.PitchBendChange:
+					var pitchBendChangeMessage = (MidiPitchBendChangeMessage)receivedMidiMessage;
+					outputMessage.Append(", Channel: ").Append(pitchBendChangeMessage.Channel).Append(", Bend: ").Append(pitchBendChangeMessage.Bend);
+					break;
+				case MidiMessageType.SystemExclusive:
+					var systemExclusiveMessage = (MidiSystemExclusiveMessage)receivedMidiMessage;
+					outputMessage.Append(", ");
+
+					// Read the SysEx bufffer
+					var sysExDataReader = DataReader.FromBuffer(systemExclusiveMessage.RawData);
+					while (sysExDataReader.UnconsumedBufferLength > 0)
+					{
+						byte byteRead = sysExDataReader.ReadByte();
+						// Pad with leading zero if necessary
+						outputMessage.Append(byteRead.ToString("X2")).Append(" ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)